notebooks/Mitigating Disparities in Ranking from Binary Data.ipynb

```diff
@@ -0,0 +1,611 @@
+    "1. [Mitigating Demographic Disparity with Grid Search](#Mitigating-Demographic-Disparity-with-Grid-Search)\n",
+    "   1. [Comparing Probabilistic Predictors using the Dashboard](#Comparing-Probabilistic-Predictors-using-the-Dashboard)\n",
+    "1. [Obtaining Low-Disparity Classifiers](#Obtaining-Low-Disparity-Classifiers)\n",
+    "   1. [Postprocessing](#Postprocessing)\n",
+    "   1. [Exponentiated Gradient](#Exponentiated-Gradient)\n",
+    "   1. [Comparing Classifiers using the Dashboard](#Comparing-Classifiers-using-the-Dashboard)\n",
+    "\n",
+    "   \n",
+    "## Introduction\n",
+    "\n",
+    "We consider the task of ranking students for admission to law school using the data collected in [Law School Admissions Council's (LSAC) National Longitudinal Bar Passage Study](https://eric.ed.gov/?id=ED469370); specifically, the version downloaded from [Project SEAPHE](http://www.seaphe.org/databases.php). We highlight some of the fairness considerations that come up not only in school admissions, but also in other ranking scenarios. Necessarily, our example is simplified and ignores many real-world considerations specific to school admissions.\n",
+    "\n",
+    "The data set contains information about law students collected by LSAC between 1991 and 1997. Some of the information is available at the admission time (such as the undergraduate GPA and LSAT score), and some describes the performance of the students once admitted. We also have access to their self-identified race. Throughout this exercise, we will limit the attention to those self-identified as **black** and **white** (two largest groups).\n",
+    "\n",
+    "To help with ranking law school applicants, we train a model that uses the information that is available about a student at the admission time to predict the probability that they will pass their bar exam. The predictions of our model are intended to be used (among other factors) by admission officers to select the applicants. After training the initial model, we examine differences in the predictions it induces across two the two groups. We then mitigate these differences using the `GridSearch` approach in `fairlearn`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data\n",
+    "\n",
+    "We download the data using the `tempeh` package, which already filters the set of students to black and white and splits them into training and test subsets. The training and test data sets are loaded in three parts:\n",
+    "\n",
+    "* **X_train**, **X_test**: features describing the training and test data; `tempeh` provides two features: `ugpa` (undegraduate GPA) and `lsat` (LSAT score)\n",
+    "\n",
+    "* **y_train**, **y_test**: labels of the training and test data; the labels are 0 or 1, indicating whether a student passed the bar exam by the 2nd attempt\n",
+    "\n",
+    "* **A_train**, **A_test**: self-identified race of each student (black or white)"
```

Comment:
I know A is shorter than sensitive_features, but we've put in so much effort to standardize this across methods. Introducing it again now feels odd.

Reply:
Sort of.... we've made sure that whenever it's declared as an argument to a method, we call the argument `sensitive_features`. This is slightly different in that it's the variable we're going to present to that method. e.g.
```python
sweep.fit(X_train, y_train, sensitive_features=A_train)
```
It is slightly inconsistent, but I'm not sure writing out `sensitive_features_train` would add anything by characters.

Reply:
i'm going to leave A for now.